### ISSUE TITLE
Sanitize during Sql-sync Fails

### ISSUE THREAD

**Issue by aweingarten** [acquia/blt#2215]
My system information:
* BLT version: 9.x

When I run this command:
```
blt sync
```

if I turn off sanitize the sync succeeds.  If it turn it back on the built fails.
```drush:
  aliases:
    remote: '${project.machine_name}.stg'
    local: self
    ci: self
  default_alias: '${drush.aliases.local}'
#  sanitize: false
modules:
```

I get the following output:
```
blt sync
> sync:db
Executing command blt 'sync:db' --define=drush.alias=self inside of Drupal VM...
[warning] The xDebug extension is loaded. This will significantly decrease performance.
> setup:settings
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/mysite/vendor/bin/drush cache-clear drush --uri=mysite.com --yes --config=/var/www/mysite/drush/drushrc.php && /var/www/mysite/vendor/bin/drush sql-drop --uri=mysite.com --yes --config=/var/www/mysite/drush/drushrc.php && /var/www/mysite/vendor/bin/drush sql-sync @mysite.stg @self --structure-tables-key=lightweight --create-db --uri=mysite.com --yes --config=/var/www/mysite/drush/drushrc.php && /var/www/mysite/vendor/bin/drush sql-sanitize --uri=mysite.com --yes --config=/var/www/mysite/drush/drushrc.php && /var/www/mysite/vendor/bin/drush cache-clear drush --uri=mysite.com --yes --config=/var/www/mysite/drush/drushrc.php in /var/www/mysite/docroot
 [success] 'drush' cache was cleared.

 // Do you really want to drop all tables in the database drupal?: yes.

You will destroy data in drupal and replace with data from mysitestg.ssh.prod.acquia-sites.com/mysitestg.
 [notice] Starting to create database on Destination.
Creating database drupal. Any existing database will be dropped!

 // Do you really want to continue?: yes.

 [notice] Starting to dump database on Source.
 [success] Database dump saved to /home/mysite/drush-backups/mysitestg/20171029182643/mysitestg_20171029_182643.sql.gz
 [notice] Command dispatch complete
 [notice] Starting to discover temporary files directory on Destination.
You will delete files in /tmp/mysitestg_20171029_182643.sql.gz and replace with data from mysite.stg@mysitestg.ssh.prod.acquia-sites.com:/home/mysite/drush-backups/mysitestg/20171029182643/mysitestg_20171029_182643.sql.gz

 // Do you want to continue?: yes.

 [notice] Copying dump file from Source to Destination.
 [notice] Starting to import dump file onto Destination database.

The following operations will be performed:

 * Truncate sessions table.
 * Sanitize text fields associated with users.
 * Sanitize user passwords.
 * Sanitize user emails.

 // Do you want to sanitize the current database?: yes.

 [success] Sessions table truncated.
 [error]  Error: Call to a member function isBaseField() on null in Drush\Drupal\Commands\sql\SanitizeUserFieldsCommands->sanitize() (line 63 of /var/www/mysite/vendor/drush/drush/src/Drupal/Commands/sql/SanitizeUserFieldsCommands.php) #0 /var/www/mysite/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/ProcessResultHookDispatcher.php(46): Drush\Drupal\Commands\sql\SanitizeUserFieldsCommands->sanitize(NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#1 /var/www/mysite/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/ProcessResultHookDispatcher.php(33): Consolidation\AnnotatedCommand\Hooks\Dispatchers\ProcessResultHookDispatcher->callProcessor(Array, NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#2 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(188): Consolidation\AnnotatedCommand\Hooks\Dispatchers\ProcessResultHookDispatcher->process(NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#3 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(182): Consolidation\AnnotatedCommand\CommandProcessor->processResults(Array, NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#4 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(150): Consolidation\AnnotatedCommand\CommandProcessor->validateRunAndAlter(Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
#5 /var/www/mysite/vendor/consolidation/annotated-command/src/AnnotatedCommand.php(404): Consolidation\AnnotatedCommand\CommandProcessor->process(Object(Symfony\Component\Console\Output\ConsoleOutput), Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
#6 /var/www/mysite/vendor/symfony/console/Command/Command.php(264): Consolidation\AnnotatedCommand\AnnotatedCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/mysite/vendor/symfony/console/Application.php(859): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /var/www/mysite/vendor/symfony/console/Application.php(206): Symfony\Component\Console\Application->doRunCommand(Object(Consolidation\AnnotatedCommand\AnnotatedCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /var/www/mysite/vendor/symfony/console/Application.php(125): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /var/www/mysite/vendor/drush/drush/src/Preflight/Preflight.php(302): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /var/www/mysite/vendor/drush/drush/src/Preflight/Preflight.php(175): Drush\Preflight\Preflight->doRun(Array)
#12 /var/www/mysite/vendor/drush/drush/drush.php(62): Drush\Preflight\Preflight->run(Array)
#13 /var/www/mysite/vendor/drush/drush/drush(4): require('/var/www/mysite/v...')
#14 {main}.
Error: Call to a member function isBaseField() on null in /var/www/mysite/vendor/drush/drush/src/Drupal/Commands/sql/SanitizeUserFieldsCommands.php on line 63 #0 /var/www/mysite/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/ProcessResultHookDispatcher.php(46): Drush\Drupal\Commands\sql\SanitizeUserFieldsCommands->sanitize(NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#1 /var/www/mysite/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/ProcessResultHookDispatcher.php(33): Consolidation\AnnotatedCommand\Hooks\Dispatchers\ProcessResultHookDispatcher->callProcessor(Array, NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#2 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(188): Consolidation\AnnotatedCommand\Hooks\Dispatchers\ProcessResultHookDispatcher->process(NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#3 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(182): Consolidation\AnnotatedCommand\CommandProcessor->processResults(Array, NULL, Object(Consolidation\AnnotatedCommand\CommandData))
#4 /var/www/mysite/vendor/consolidation/annotated-command/src/CommandProcessor.php(150): Consolidation\AnnotatedCommand\CommandProcessor->validateRunAndAlter(Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
#5 /var/www/mysite/vendor/consolidation/annotated-command/src/AnnotatedCommand.php(404): Consolidation\AnnotatedCommand\CommandProcessor->process(Object(Symfony\Component\Console\Output\ConsoleOutput), Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
#6 /var/www/mysite/vendor/symfony/console/Command/Command.php(264): Consolidation\AnnotatedCommand\AnnotatedCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/mysite/vendor/symfony/console/Application.php(859): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /var/www/mysite/vendor/symfony/console/Application.php(206): Symfony\Component\Console\Application->doRunCommand(Object(Consolidation\AnnotatedCommand\AnnotatedCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /var/www/mysite/vendor/symfony/console/Application.php(125): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /var/www/mysite/vendor/drush/drush/src/Preflight/Preflight.php(302): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /var/www/mysite/vendor/drush/drush/src/Preflight/Preflight.php(175): Drush\Preflight\Preflight->doRun(Array)
#12 /var/www/mysite/vendor/drush/drush/drush.php(62): Drush\Preflight\Preflight->run(Array)
#13 /var/www/mysite/vendor/drush/drush/drush(4): require('/var/www/mysite/v...')
#14 {main}
Error: Call to a member function isBaseField() on null in Drush\Drupal\Commands\sql\SanitizeUserFieldsCommands->sanitize() (line 63 of /var/www/mysite/vendor/drush/drush/src/Drupal/Commands/sql/SanitizeUserFieldsCommands.php).
 [error]  Drush command terminated abnormally due to an unrecoverable error.
[Acquia\Blt\Robo\Tasks\DrushTask]  Exit code 1  Time 8.701s
[error]
Connection to 127.0.0.1 closed.
```

**Comment by danepowell** [comment]
Seems like an upstream bug:
https://github.com/drush-ops/drush/pull/3070

Looks like it was fixed in Drush 9.0.0-beta8, can you confirm?

**Comment by danepowell** [comment]
Closing due to inactivity
